st2common/st2common/services/sensor_watcher.py

```diff
@@ -95,9 +95,18 @@ def start(self):
             self.connection.release()
 
     def stop(self):
+        LOG.debug('Shutting down sensor watcher.')
         try:
             if self._updates_thread:
                 self._updates_thread = eventlet.kill(self._updates_thread)
+
+            if self.connection:
+                channel = self.connection.channel()
```

Comment:
The sensor watcher isn't tied to the sensors life cycle at all. See code here https://github.com/StackStorm/st2/blob/master/st2reactor/st2reactor/container/manager.py#L66. From my reading, it seems like the stop code path will be invoked when the st2sensorcontainer process is shutdown by user running sudo service/systemctl stop or killing the process with Ctrl + C. So I think it's safe to delete the queue here. I can try with durable and see if that works really. I tested the explicit delete I have in dev env and it definitely works.